components/cldr-json-data-provider/src/transform/dates.rs

```diff
@@ -28,7 +28,8 @@ impl TryFrom<&dyn CldrPaths> for DatesProvider<'_> {
         for dir in locale_dirs {
             let path = dir.join("ca-gregorian.json");
 
-            let mut resource: cldr_json::Resource = serde_json::from_reader(open_reader(path)?)?;
+            let mut resource: cldr_json::Resource =
+                serde_json::from_reader(open_reader(&path)?).map_err(|e| (e, Some(path)))?;
```

Comment:
If you like this pattern for attaching extra information to an error, I'll start using it elsewhere, too.  I transform `e` to `(e, extra_info)`, and then I have `impl From<(ChildError, ExtraInfo)> for Error`.

Reply:
yeah, it looks good to me! Maybe we could use it with FieldLength errors in DateTime later!